web3/eth.py

```diff
@@ -211,6 +211,13 @@ def getTransactionFromBlock(self, block_identifier, transaction_index):
             [block_identifier, transaction_index],
         )
 
+    def getTransactionByBlock(self, block_identifier, transaction_index):
+        """
+        Alias for the method getTransactionFromBlock
+        More details here: https://github.com/ethereum/web3.py/pull/857
+        """
+        return self.getTransactionFromBlock(block_identifier, transaction_index)
```

Comment:
I think it's cleaner to put the implementation in this new method, and alias from the deprecated version to this implementation . Then removing the deprecated name in v5 will be as simple as deleting the deprecated function.

Reply:
Agreed, have made the changes